includes/discount-functions.php

```diff
@@ -32,19 +32,29 @@ function edd_get_discounts( $args = array() ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$discounts = get_posts( $args );
+	$discounts_hash = md5( json_encode( $args ) );
+	if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {
+		$discounts = get_posts( $args );
+		wp_cache_set( $discounts_hash, $discounts, 'edd-discounts' );
+	}
+
 
 	if ( $discounts ) {
 		return $discounts;
 	}
 
+	// If no discounts are found and we are searching, re-query with a meta key to find discounts by code
 	if( ! $discounts && ! empty( $args['s'] ) ) {
-		// If no discounts are found and we are searching, re-query with a meta key to find discounts by code
 		$args['meta_key']     = '_edd_discount_code';
 		$args['meta_value']   = $args['s'];
 		$args['meta_compare'] = 'LIKE';
 		unset( $args['s'] );
-		$discounts = get_posts( $args );
+
+		$discounts_hash = md5( json_encode( $args ) );
+		if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {
```

Comment:
In order to maintain compatibility with PHP 5.3, please declare and call the function outside the `if` statement and then reference the variable.

i.e.
```php
$discounts = wp_cache_get( $discounts_hash, 'edd-discounts' )
if ( false === $discounts ) {
```